### DISCUSSION TITLE
React Query Invalidation Does Not Trigger Re-Render During Cypress Component Test

### DISCUSSION THREAD

**Discussion by matsch01** [TanStack/query#3607] [Q&A]
I have a Cypress component test that depends on a React Query invalidation. However, invalidating a query triggers no re-renders during the test. The invalidation works fine when running normally.

Is this a known issue/feature? 

@cypress/react: 5.12.4
react: 17.0.2
react-query: 3.34.8

**Update**

I found a workaround here:

[Workaround](https://stackoverflow.com/questions/69344045/how-to-access-the-react-query-queryclient-outside-the-provider-e-g-to-invali)

Still I would like to know why this does not work normally during the test.

**Comment by TkDodo** [comment]
no, it's not. please show some form of reproduction. cypress is just a browser, so it should work like any other browser. We run our tests in cypress without problems.

**Reply by matsch01** [reply]
I figured out my problem. I wasn't passing the client to QueryClientProvider properly when mounting my component. It's all working well now.
